File: app/src/main/java/fragments/NotificationFragment.java
package fragments;


import android.os.Bundle;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import androidx.swiperefreshlayout.widget.SwipeRefreshLayout;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.google.firebase.auth.FirebaseUser;
import com.google.firebase.firestore.Query;
import com.sqube.tipshub.R;

import adapters.NotificationAdapter;
import utils.FirebaseUtil;


/**
 * A simple {@link Fragment} subclass.
 */
public class NotificationFragment extends Fragment {


    public NotificationFragment() {
        // Required empty public constructor
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {

        // Inflate the layout for this fragment
        View rootView=inflater.inflate(R.layout.fragment_notification, container, false);
        RecyclerView notificationList = rootView.findViewById(R.id.postList);
        notificationList.setLayoutManager(new LinearLayoutManager(getActivity()));
        SwipeRefreshLayout refresher = rootView.findViewById(R.id.refresher);
        refresher.setColorSchemeResources(R.color.colorPrimary);

        FirebaseUser user = FirebaseUtil.getFirebaseAuthentication().getCurrentUser();
        String userId = user.getUid();
        Query query = FirebaseUtil.getFirebaseFirestore().collection("notifications")
                .orderBy("time", Query.Direction.DESCENDING).whereEqualTo("sendTo", userId).limit(40);

        NotificationAdapter notificationAdapter = new NotificationAdapter(query, userId, getContext());
        notificationList.setAdapter(notificationAdapter);
        notificationAdapter.startListening();

        refresher.setOnRefreshListener(() -> {
            refresher.setRefreshing(true);
            if(notificationAdapter !=null) {
                notificationAdapter.stopListening();
                notificationAdapter.startListening();
            }
            refresher.setRefreshing(false);
        });
        return rootView;
    }
}

File: app/src/main/java/fragments/NotificationFragment.kt
package fragments

import adapters.NotificationAdapter
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.LinearLayoutManager
import com.google.firebase.firestore.Query
import com.sqube.tipshub.R
import com.sqube.tipshub.databinding.FragmentNotificationBinding
import utils.FirebaseUtil

/**
 * A simple [Fragment] subclass.
 */
class NotificationFragment : Fragment() {
    private lateinit var _binding: FragmentNotificationBinding
    private val binding get() = _binding

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View {
        _binding = FragmentNotificationBinding.inflate(inflater, container, false)

        binding.postList.layoutManager = LinearLayoutManager(activity)
        binding.refresher.setColorSchemeResources(R.color.colorPrimary)
        val user = FirebaseUtil.firebaseAuthentication?.currentUser
        val userId = user!!.uid
        val query = FirebaseUtil.firebaseFirestore?.collection("notifications")!!
                .orderBy("time", Query.Direction.DESCENDING).whereEqualTo("sendTo", userId).limit(40)
        val notificationAdapter = NotificationAdapter(query, userId)
        binding.postList.adapter = notificationAdapter
        notificationAdapter.startListening()
        binding.refresher.setOnRefreshListener {
            binding.refresher.isRefreshing = true
            notificationAdapter.stopListening()
            notificationAdapter.startListening()
            binding.refresher.isRefreshing = false
        }
        return binding.root
    }
    fun scrollToTop(){
        binding.postList.smoothScrollToPosition(0)
    }
}